### ISSUE TITLE
Detect named functions 

### ISSUE THREAD

**Issue by vincentbernat** [antonmedv/expr#317]
After updating to 1.10:

```
interface conversion: interface {} is core.classifyStringFunc, not func(string) bool (1:1)
```

However, I have:

```go
type classifyStringFunc func(string) bool
```

I can work around the issue with

```go
type classifyStringFunc = func(string) bool
```

_Originally posted by @vincentbernat in https://github.com/antonmedv/expr/issues/310#issuecomment-1401082756_

**Comment by antonmedv** [comment]
Found the issue: problem with new optimization: it tries to cast to one of the known func types. But in this case this is not doable and reflection call type should be used.

**Comment by antonmedv** [comment]
Fixed in cffbb1f

**Comment by vincentbernat** [comment]
Thanks for the quick fix! I can confirm it works for me too. I'll keep the workaround as it would avoid using reflection.
